### ISSUE TITLE
Multi-process file signature checking

### ISSUE THREAD

**Issue by gaow** [vatlab/SoS#454]
This is related to #450. I noticed the file signature check is performed with a single thread. Is that the case? Would it be helpful to use multiprocessing for the task?

**Comment by BoPeng** [comment]
Signature checking is done within each step, which is multi-processing if steps are executed in parallel. The signature writing or checking process can be parallelized (e.g. calculating of md5 of multiple files using a thread pool) but that is only helpful when a large number of files are involved. 

WONTFIX unless there is a good user case for optimization.
